examples/LEDs/main.c

```diff
@@ -0,0 +1,328 @@
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED0_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED0_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED1_IS_PRESENT
+static int led1_command(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED1_ON;
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED1_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED1_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED2_IS_PRESENT
+static int led2_command(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED2_ON;
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED2_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED2_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED3_IS_PRESENT
+static int led3_command(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED3_ON;
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED3_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED3_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED4_IS_PRESENT
+static int led4_command(int argc, char **argv)
+{
+    if (argc <= 1) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED4_ON;
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED4_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED4_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED5_IS_PRESENT
+static int led5_command(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED3_ON;
+    }
+    else if (strcmp(argv[1], "off") == 0) {
+        LED3_OFF;
+    }
+    else if (strcmp(argv[1], "toggle") == 0) {
+        LED3_TOGGLE;
+    }
+    else {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+    }
+
+    return 0;
+}
+#endif
+
+#ifdef LED6_IS_PRESENT
+static int led6_command(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <on|off|toggle>\n", argv[0]);
+        return -1;
+    }
+
+    if (strcmp(argv[1], "on") == 0) {
+        LED3_ON;
```

Comment:
copy-paste issues here and below :) also I think it's a bit unfortunate to have that much code-duplication in a beginner-targeted example. How about having a command like `led <id> <on|off|toggle>` using [`LED_ON(6)`](https://doc.riot-os.org/group__drivers__led.html#ga002d6bac2cf4e6510ebcef63eaef36ae) instead?

Reply:
Copy-paste mistake corrected.

I know that there is much copy-paste code - but in my opinion for newbie finding how many LEDs are available could be a little tricky. In this, maybe not so nice solution, information about number of LEDs is clear. And the code is rather simple.

Reply:
If you are mainly concerned about providing the number of available LEDs to the user, why not checking and printing the number on startup, using a dedicated command or, on request, when the `led` command would be used with a non-existent LED?

Reply:
I still have some doubts about complications of the code. Moreover, `LED_ON(x)` is a macro and simply we cannot call it with LED id from variable `LED_ON(id)` but there should be added code like:

```
 switch(id) {
   case 1: LED_ON(1); break;
   case 2: LED_ON(2); break;
   ....
}
```

and the code for counting LEDs have to repeat some code like:

```
#ifdef LED0_IS_PRESENT
led_no++;
#endif
#ifdef LED1_IS_PRESENT
led_no++;
#endif
```

If you think that this is a better solution, I could reorganize code.

Reply:
Hmm true, the macro doesn't help. In a perfect world, I would expect some static inline functions `led_{on,off,toggle}` provided in the `led.h` header that abstract such switch-case away for users. Not sure if it was a deliberate decision to not provide such functions until now.

As to the LED counting code, according to [the documentation](https://doc.riot-os.org/group__drivers__led.html), the present LEDs should be defined consecutively starting from `LED0`, so instead of actually counting, you could check for the highest available LED number.

But again, that information is something that could (and probably should) be provided by an additional macro in `led.h` instead.

Reply:
See #20783 for an implementation of both ideas.

Reply:
Thanks for addition of these features. I refactor code as soon as PR #20783 will be merged to master.